Implement batch processing workflow

SayantikaPaul-12 opened the pull request:
- Add comprehensive user profile extraction in batch processor
- Enhance SQS processor with AgentCore integration and streaming response handling
- Fix IAM permissions for DynamoDB access in CDK stack
- Add proper session ID generation (33+ chars for AgentCore compatibility)
- Improve error handling and logging throughout the pipeline
- Add Docker configuration for SQS processor with proper dependencies
- Create deployment script for easy CDK deployment
- Update DynamoDB tools with enhanced job recommendation saving

Key improvements:
- Batch processor now extracts full user profiles (name, location, job preferences)
- SQS processor creates personalized job search prompts using user data
- Fixed AgentCore response parsing for streaming data
- Added comprehensive comments and documentation
- Enhanced IAM roles and permissions for AgentCore integration

SayantikaPaul-12 committed:
feat: implement enhanced batch processing workflow

- Add comprehensive user profile extraction in batch processor
- Enhance SQS processor with AgentCore integration and streaming response handling
- Fix IAM permissions for DynamoDB access in CDK stack
- Add proper session ID generation (33+ chars for AgentCore compatibility)
- Improve error handling and logging throughout the pipeline
- Add Docker configuration for SQS processor with proper dependencies
- Create deployment script for easy CDK deployment
- Update DynamoDB tools with enhanced job recommendation saving

Key improvements:
- Batch processor now extracts full user profiles (name, location, job preferences)
- SQS processor creates personalized job search prompts using user data
- Fixed AgentCore response parsing for streaming data
- Added comprehensive comments and documentation
- Enhanced IAM roles and permissions for AgentCore integration

aryankhanna2004 commented on backend/lib/backend-stack.ts:
```diff
@@ -153,22 +153,92 @@ export class jobsearch1 extends cdk.Stack {
+    // const jobSearchAgentImage = new ecrAssets.DockerImageAsset(this, 'JobSearchAgentImage', {
+    //   directory: path.join(__dirname, '..', 'JobSearchAgent'),
+    //   platform: lambdaArchitecture === lambda.Architecture.ARM_64
+    //     ? ecrAssets.Platform.LINUX_ARM64
```
This image is needed as this is the core strands agents

aryankhanna2004 commented on backend/lib/backend-stack.ts:
```diff
@@ -153,22 +153,92 @@ export class jobsearch1 extends cdk.Stack {
+          statements: [
+            new iam.PolicyStatement({
+              effect: iam.Effect.ALLOW,
+              actions: [
```
We probably dont need this role as it is not getting used anywhere

aryankhanna2004 commented on backend/lib/backend-stack.ts:
```diff
@@ -209,23 +281,31 @@ export class jobsearch1 extends cdk.Stack {
+        BEDROCK_AGENTCORE_QUALIFIER: 'DEFAULT',
+        STUDENT_PROFILE_TABLE_NAME: StudentProfileTable.tableName,
+        JOB_RECOMMENDATIONS_TABLE_NAME: JobRecommendationsTable.tableName,
+        JOB_SEARCH_KB: kb.knowledgeBaseId,
```
please check once what variables you are using in this lambda function. Like it probably doesnt need job search kb tables memory id user preferences etc

aryankhanna2004 commented on backend/fix-and-deploy.sh:
```diff
@@ -0,0 +1,63 @@
+
+if [ "$STACK_STATUS" = "UPDATE_ROLLBACK_COMPLETE" ]; then
+    echo "⚠️  Stack is in UPDATE_ROLLBACK_COMPLETE state"
+    echo "🗑️  Deleting the stuck stack..."
```
Can you please delete this file?

aryankhanna2004 commented on backend/lib/backend-stack.ts:
```diff
@@ -275,13 +355,14 @@ export class jobsearch1 extends cdk.Stack {
+
+    // Skip Docker image output for faster deployment
+    // new cdk.CfnOutput(this, 'DockerImageURI', {
+    //   value: jobSearchAgentImage.imageUri,
```
Can you please bring this back?

aryankhanna2004 committed:
refactor: remove deprecated fix-and-deploy script and update job recommendation saving structure

- Deleted the outdated fix-and-deploy.sh script to streamline deployment processes.
- Updated save_job_recommendations function in dynamodb_tools.py to accept structured jobInformation parameter instead of job_ids and sent_via.
- Enhanced job search agent workflow in StrandsAgents.py to include detailed job role extraction and comprehensive fit analysis.
- Adjusted SQS processor and batch processor to improve user profile handling and job matching efficiency.
- Updated CDK stack to reflect changes in environment variables and IAM roles for better integration.

aryankhanna2004 committed:
fix: enhance session ID generation for compatibility

- Added validation to regenerate session ID if it is shorter than 33 characters for backward compatibility.
- Improved session ID generation logic to ensure it meets the minimum length requirement by concatenating additional random characters if necessary.
- Updated comments for clarity on session ID handling and storage in sessionStorage.
